### PR TITLE
CONSOLE-2838:  masthead changes to better align with ACM

### PR DESCRIPTION
**Pull request by rhamilto**
https://github.com/openshift/console/pull/9508 includes the about modal changes since those depend on ACM changes that are still under development.

Changes in this PR:
* If ACM is installed, `ACM Documentation` is added to the help menu
* Remove the padding override around the username dropdown toggle as it is no longer needed

### CODE REVIEW

File: frontend/@types/console/index.d.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ declare interface Window {
     consolePlugins: string[]; // Console dynamic plugins enabled on the cluster
     quickStarts: string;
     projectAccessClusterRoles: string;
+    multicluster: boolean;
```
Comment (spadgett) [inline frontend/@types/console/index.d.ts:49]: We'll want to align with how we test this in #9340. Probably you'll want to reuse `isMultiClusterEnabled`.

https://github.com/openshift/console/pull/9340/files#diff-52414ab87aedb2a2b855ef5f11c0d85c63286f32c74831d1edb730919c3fc220R1

cc @abhinandan13jan @christianvogt

File: frontend/public/components/about-modal.tsx [hunk 2]
```diff
@@ -74,6 +77,14 @@ const AboutModalItems: React.FC<AboutModalItemsProps> = ({ closeAboutModal }) =>
               </TextListItem>
             </>
           )}
+          {acmVersion && (
```
Comment (spadgett) [inline frontend/public/components/about-modal.tsx:80]: We might need to make this part of another PR, or we'd need to hold this PR a few sprints. CONSOLE-2840 has dependencies on the ACM team and won't be finished soon.

File: frontend/public/components/masthead-toolbar.jsx [hunk 3]
```diff
@@ -318,13 +319,27 @@ class MastheadToolbarContents_ extends React.Component {
           component: <Link to="/quickstart">{t('public~Quick Starts')}</Link>,
         },
         {
-          label: t('public~Documentation'),
+          label: multicluster ? t('public~OCP Documentation') : t('public~Documentation'),
```
Comment (spadgett) [inline frontend/public/components/masthead-toolbar.jsx:322]: We'll need to adjust the label here based on the branding (OKD/OCP/OSD/etc).

**Review by spadgett — COMMENTED** [review]
Thanks, @rhamilto.

File: frontend/@types/console/index.d.ts (hunk 1 above)
Comment (rhamilto) [reply frontend/@types/console/index.d.ts:49]: Is there a case where ACM is installed, but there is only one cluster?  https://github.com/openshift/console/pull/9340/files#diff-5f2cc074d9cdf86072a6e8caf63290f7d49b2e126b2c30ff3159bd36923dfff6R9 

Or does ACM always require more than one cluster?

I ask as the switching in this PR is specific to the presence of ACM and not necessarily multicluster.

Comment (spadgett) [reply frontend/@types/console/index.d.ts:49]: @rhamilto I think you could check for an empty array, which would mean the ACM integration is not configured. If ACM has no managed clusters, you'd get a list of size one with `local-cluster`.

@TheRealJon can you confirm?

Comment (TheRealJon) [reply frontend/@types/console/index.d.ts:49]: We haven't implemented the back end changes yet, but I can make it so.

Comment (christianvogt) [reply frontend/@types/console/index.d.ts:49]: @spadgett i thought you said that we'd always at least have a single item in the array. If with ACM integration or not, we'd have `local-cluster` present.
I think having a `null` value is fine to with no ACM integration though

Comment (rhamilto) [reply frontend/@types/console/index.d.ts:49]: Looking to close the loop on this...  What is the correct check for ACM configuration?   I am unclear on what the correct value of `window.SERVER_FLAGS.clusters` will be.  Is `local-cluster` always present in the array or not?

File: frontend/public/components/masthead-toolbar.jsx (hunk 3 above)
Comment (rhamilto) [reply frontend/public/components/masthead-toolbar.jsx:322]: Wondering if it makes sense to leave this simply labelled `Documentation` and differentiate the addition of ACM with the planned `ACM Documentation` so we don't have to do such complex switching with the product name?  What do you think @andybraren, @bmignano?

Comment (andybraren) [reply frontend/public/components/masthead-toolbar.jsx:322]: Good catch. `Documentation` and `ACM Documentation` works, although I wonder if `OpenShift Documentation` would be even clearer while still being generic enough to cover all three.

By the way, does this link point to the [OKD Docs](https://docs.okd.io/latest/welcome/index.html) when the Console is running on OKD, or does it always point to OpenShift's docs today?

Comment (rhamilto) [reply frontend/public/components/masthead-toolbar.jsx:322]: > By the way, does this link point to the [OKD Docs](https://docs.okd.io/latest/welcome/index.html) when the Console is running on OKD, or does it always point to OpenShift's docs today?

For OKD, the link points to `https://docs.okd.io/latest/`, so we'd still have to differentiate between OpenShift and OKD.  Leaving the label as `Documentation` for all eliminates the need to do *any* label switching.

File: frontend/@types/console/index.d.ts (hunk 1 above)
Comment (TheRealJon) [reply frontend/@types/console/index.d.ts]: IMO, `window.SERVER_FLAGS.clusters` should always contain the cluster that the console is running on, so it would never be empty.

File: frontend/public/components/about-modal.tsx (hunk 2 above)
Comment (rhamilto) [reply frontend/public/components/about-modal.tsx]: About modal changes moved to https://github.com/openshift/console/pull/9508.

File: frontend/public/components/masthead-toolbar.jsx (hunk 3 above)
Comment (andybraren) [reply frontend/public/components/masthead-toolbar.jsx]: Works for me.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by spadgett** [comment]
@rhamilto @yapei It will be hard for QE to verify the ACM documentation link since we haven't merged the operator and backend changes for the ACM integration. We could simply confirm there are no regressions.

@yapei You might be able to set `window.SERVER_FLAGS.clusters = ['foo','bar']` in the browser JS console to test the link.

**Comment by yapei** [comment]
The ACM operator installation is blocked(always in Pending status) due to legacy apiVersion in CRD (tracked in https://bugzilla.redhat.com/show_bug.cgi?id=1981227 )

Under this circumstance, I did 

1. Created `consolelink/acm-console` resource manually and it has following spec
```
apiVersion: console.openshift.io/v1
kind: ConsoleLink
metadata:
  name: acm-console-link
spec:
  applicationMenu:
    imageURL: https://CONSOLE_ROUTE/multicloud/header/graphics/RHACM-logo_24x24.svg
    section: Red Hat Applications
  href: https://CONSOLE_ROUTE
  location: ApplicationMenu
  text: Red Hat Advanced Cluster Management for Kubernetes
```
2. Set `window.SERVER_FLAGS.clusters = ['foo','bar']` in browser JS console

At last, we check the `Advanced Cluster Management` appears in perspective dropdown(regression) and `ACM Documentation` link

**Comment by yapei** [comment]
The ACM operator installation is blocked(always in Pending status) due to legacy apiVersion in CRD (tracked in https://bugzilla.redhat.com/show_bug.cgi?id=1981227 )

Under this circumstance, I did 

1. Created `consolelink/acm-console` resource manually and it has following spec
```
apiVersion: console.openshift.io/v1
kind: ConsoleLink
metadata:
  name: acm-console-link
spec:
  applicationMenu:
    imageURL: https://CONSOLE_ROUTE/multicloud/header/graphics/RHACM-logo_24x24.svg
    section: Red Hat Applications
  href: https://CONSOLE_ROUTE
  location: ApplicationMenu
  text: Red Hat Advanced Cluster Management for Kubernetes
```
2. Set `window.SERVER_FLAGS.clusters = ['foo','bar']` in browser JS console

At last, we check the `Advanced Cluster Management` appears in perspective dropdown(regression) and `ACM Documentation` link is shown in Helper dropdown list, clicking it will redirect to https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.2/

**Comment by yapei** [comment]
no user visible changes for the `_overrides.scss` update, no issues found. 
Let me know if I'm testing wrongly
/label qe-approved

**Comment by rhamilto** [comment]
@yapei, step 2 is the relevant one for the changes in this PR.  You can do step 1, but nothing in the PR will impact the application launcher.

**Comment by rhamilto** [comment]
> no user visible changes for the `_overrides.scss` update, no issues found.

You should see the user dropdown toggle shift left.

https://user-images.githubusercontent.com/895728/125791271-b01e45d1-885f-43f0-9d4e-ad529281c002.mov

**Comment by yapei** [comment]
> > no user visible changes for the `_overrides.scss` update, no issues found.
> 
> You should see the user dropdown toggle shift left.
> Screen.Recording.2021-07-15.at.8.52.04.AM.mov

Thank you!

**Comment by rhamilto** [comment]
@sferich888, please review as this PR is awaiting `px-approved`.
